pymc/distributions/logprob.py

```diff
@@ -141,8 +141,109 @@ def logpt(
         A scaling term to apply to the generated log-likelihood graph.
     transformed
         Apply transforms.
-    cdf
-        Return the log cumulative distribution.
+
+    """
+
+    if rv_values is None:
+        if isinstance(var.owner.op, RandomVariable):
+            rv_value_var = getattr(var.tag, "observations", getattr(var.tag, "value_var", var))
+            rv_values = {var: rv_value_var}
+        else:
+            rv_values = {}
+            raise Warning("No value variables provided the logp will be an empty graph")
+    elif not isinstance(rv_values, Mapping):
+        rv_values = {var: at.as_tensor_variable(rv_values).astype(var.type)}
+
+    tmp_rvs_to_values = rv_values.copy()
+    transform_map = {}
+    for node in io_toposort(graph_inputs((var,)), (var,)):
+        if isinstance(node.op, RandomVariable):
+            curr_var = node.out
+            rv_value_var = getattr(
+                curr_var.tag, "observations", getattr(curr_var.tag, "value_var", curr_var)
+            )
+            rv_value = rv_values.get(curr_var, rv_value_var)
+            tmp_rvs_to_values[curr_var] = rv_value
+            # if hasattr(rv_value_var.tag, "transform") and transformed:
+            #     transform_map[rv_value] = rv_value_var.tag.transform
+        # The condition below is a hackish way of excluding the value variable for the
+        # RV being indexed in case of Advanced Indexing of RVs. It gets added by the
+        # logic above but aeppl does not expect us to include it in the dictionary of
+        # {RV:values} given to it.
+        if isinstance(node.op, subtensor_types):
+            curr_var = node.out
+            if (
+                curr_var in tmp_rvs_to_values.keys()
+                and curr_var.owner.inputs[0] in tmp_rvs_to_values.keys()
+            ):
+                tmp_rvs_to_values.pop(curr_var.owner.inputs[0])
+
+    transform_opt = TransformValuesOpt(transform_map)
+    temp_logp_var_dict = factorized_joint_logprob(
+        tmp_rvs_to_values, extra_rewrites=transform_opt, use_jacobian=jacobian, **kwargs
+    )
+
+    logp_var_dict = {}
+    for value_var, _logp in temp_logp_var_dict.items():
+        if value_var in rv_values.values():
+            logp_var_dict[value_var] = _logp
+
+    if not logp_var_dict:
+        logp_var = None
+
```

Comment:
This particular case was because `factorized_joint_logprob` returns an empty dictionary when there are no logp graph where-as PyMC expects it to be `None`. Hence this shortcut. Regardless you're right, I should probably add such information as comments.